### ISSUE TITLE
aws-fis: CfnExperimentTemplate does not generate a valid AWS::FIS::ExperimentTemplate

### ISSUE THREAD

**Issue by jwoehrle** [aws/aws-cdk#14310]
CFN Yaml synthesized from `CfnExperimentTemplate`does not create a correct CFN `AWS::FIS::ExperimentTemplate`.

All keys in the synth are lower case, however in CFN they are expected as uppercase. This ends in errors during deploy like

```
Properties validation failed for resource fistemplate with message:
#: required key [Description] not found
#: required key [StopConditions] not found
#: required key [Targets] not found
#: required key [RoleArn] not found
#: required key [Tags] not found
#: extraneous key [stopConditions] is not permitted
#: extraneous key [description] is not permitted
#: extraneous key [targets] is not permitted
#: extraneous key [actions] is not permitted
#: extraneous key [roleArn] is not permitted
#: extraneous key [tags] is not permitted
```

### Environment

  - **CDK CLI Version:** 
  ❯ cdk --version
1.100.0 (build d996c6d)
  
  - **Module Version:** 
 1.100.0
  - **Node.js Version:** 
  ❯ node -v
v14.15.4
  - **OS:** 
  OSX Big Sur
  - **Language:** 
  TypeScript

### Other information
I'm using the following CDK snippet:

```
new fis.CfnExperimentTemplate(this,'fis-template', {
      description: 'My first Template',
      roleArn: role.roleArn,
      stopConditions: [
        { source: 'none' }
      ],
      tags: {},
      actions: {
        'TerminateInstances' : // omitted for brevity
      },
      targets: {
        'myInstance' : // omitted for brevity
      }
    })
```

deploying this fails with the above error message.
Looking at the synth output:
```
  fistemplate:
    Type: AWS::FIS::ExperimentTemplate
    Properties:
      description: My first Template
      roleArn:
        Fn::GetAtt:
          - fisrole33E76559
          - Arn
      stopConditions:
        - source: none
      tags: {}
      targets:
        myInstance:
          resourceTags: {}
          resourceType: aws:ec2:instance
          selectionMode: COUNT(1)
      actions:
        TerminateInstances:
          actionId: aws:ec2:stop-instances
          parameters: {}
          targets: {}
```

you can see that the property-keys like `description`, `roleArn`, ... are all starting with a lower letter. This is not what CFN expects according to https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-fis-experimenttemplate.html.
If I manually change the synth output to uppercase (and fix some other issues, which I created #14309 for) and then deploy via `aws cloudformation create-stack ...` I can work around this issue.

--- 

This is :bug: Bug Report

**Comment by peterwoodworth** [comment]
Thank you for submitting this issue @jwoehrle!

This issue lies with CFN and our auto-generation of L1s. It seems like this is a breaking change published in the latest version of the `cfnSpec`, we will reach out to the CFN service team to make sure it was intentional. In any case, this will get resolved when the new spec version is merged to the CDK (usually a few days)

**Comment by jwoehrle** [comment]
It seems this is related to #14309:

I use the CDK escape hatches `addPropertyDeletionOverride` and `addPropertyOverride` to fix the wrong casing (basically deleting the lowercase key, then re-adding with uppercase).

Once I had this fix in place in CDK the params got populated correctly again.

**Comment by peterwoodworth** [comment]
Yeah I suspected that the other issue you submitted was related. Glad you were able to figure out a workaround for yourself in the meantime before the new spec gets merged, and by doing so you could confirm my suspicion 😄

**Comment by mrmilosz** [comment]
https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-fis/README.md states:

> "All classes with the Cfn prefix in this module (CFN Resources) are always stable and safe to use."

which is not the case here. Should the "stable" label be removed until there is some kind of automated testing that ensures the stability of this module? For example, that it generates valid statements?

**Comment by skinny85** [comment]
Hi @mrmilosz,

Let me explain. The classes with the `Cfn*` prefix in the CDK are generated from the [CloudFormation Resource Specification](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html).

In some cases, the Resource Specification contains mistakes (meaning, what is in the Resource Spec is _not_ what is expected on the server side). This is what is happening here. In those rare cases, the Resource Specification is adjusted to correctly represent the expected shape of the resources. After that happens, when the CDK ingests the new version of the Specification with the breaking changes, it can result in breaking changes in your code if you used the previous version of the classes.

However, since the previous classes would not work when performing the CloudFormation deployment anyway, we consider these breaking changes acceptable - basically, there is no alternative in this case, because it is the server-side APIs that have changed here. Hiding those breaking changes would simply delay you finding out about the problem until deployment time. In our opinion, it's much better if you find them out at compile time.

Let me emphasize that these breaking changes happen **only** when there is a mistake in the Resource Specification. That is a rare occurrence that only happens for brand-new resources. After the Resource Specification has been fixed, the classes will then be 100% stable.

Hoe this clears this up!

Thanks,
Adam

**Comment by mrmilosz** [comment]
Hi @skinny85,

I see! I appreciate the thorough reply. I did not realize the code was being generated from a spec like that. Indeed I noticed that the fis.generated.js file incorrectly contained lowercase keys, but did not know what the source of that generated code was. Thanks for clearing that up!

I guess at this point my options are to either figure out how to temporarily generate my own code from the spec, or to apply the workarounds mentioned above (property overrides) and wait for an update to the specification to get merged in.

Sorry for derailing the discussion! :D
